includes/Core/McpAdapter.php

```diff
@@ -54,146 +33,51 @@ class McpAdapter {
 	 * Create and register a new MCP server.
 	 *
-	 * @param string        $server_id Unique identifier for the server.
+	 * @param string        $server_id              Unique identifier for the server.
 	 * @param string        $server_route_namespace Server route namespace.
-	 * @param string        $server_route Server route.
-	 * @param string        $server_name Server name.
-	 * @param string        $server_description Server description.
-	 * @param string        $server_version Server version.
-	 * @param array         $mcp_transports Array of classes that extend the BaseTransport.
-	 * @param string|null   $error_handler The error handler class name. If null, NullMcpErrorHandler will be used.
-	 * @param string|null   $observability_handler The observability handler class name. If null, NullMcpObservabilityHandler will be used.
+	 * @param string        $server_route           Server route.
+	 * @param string        $server_name            Server name.
+	 * @param string        $server_description     Server description.
+	 * @param string        $server_version         Server version.
+	 * @param array         $mcp_transports         Array of classes that extend the BaseTransport.
+	 * @param ?class-string<\WP\MCP\Infrastructure\ErrorHandling\Contracts\McpErrorHandlerInterface>         $error_handler The error handler class name. If null, NullMcpErrorHandler will be used.
+	 * @param ?class-string<\WP\MCP\Infrastructure\Observability\Contracts\McpObservabilityHandlerInterface> $observability_handler The observability handler class name. If null, NullMcpObservabilityHandler will be used.
 	 * @param array         $tools Ability names to register as tools.
 	 * @param array         $resources Resources to register.
 	 * @param array         $prompts Prompts to register.
 	 * @param callable|null $transport_permission_callback Optional custom permission callback for transport-level authentication. If null, defaults to is_user_logged_in().
 	 *
-	 * @return \WP\MCP\Core\McpAdapter
+	 * @return true|\WP_Error True on success, WP_Error on failure.
 	 * @throws \Exception If the server already exists or if called outside of the mcp_adapter_init action.
 	 */
-	public function create_server( string $server_id, string $server_route_namespace, string $server_route, string $server_name, string $server_description, string $server_version, array $mcp_transports, ?string $error_handler, ?string $observability_handler = null, array $tools = array(), array $resources = array(), array $prompts = array(), ?callable $transport_permission_callback = null ): self {
+	public function create_server( string $server_id, string $server_route_namespace, string $server_route, string $server_name, string $server_description, string $server_version, array $mcp_transports, ?string $error_handler, ?string $observability_handler = null, array $tools = array(), array $resources = array(), array $prompts = array(), ?callable $transport_permission_callback = null ) {
```

Comment:
This leaked for a different PR.

```suggestion
	 * @return \WP\MCP\Core\McpAdapter
	 * @throws \Exception If the server already exists or if called outside of the mcp_adapter_init action.
	 */
	public function create_server( string $server_id, string $server_route_namespace, string $server_route, string $server_name, string $server_description, string $server_version, array $mcp_transports, ?string $error_handler, ?string $observability_handler = null, array $tools = array(), array $resources = array(), array $prompts = array(), ?callable $transport_permission_callback = null ): self {
```